hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/scope/HandleHederaNativeOperations.java

```diff
@@ -122,9 +121,6 @@ public void setNonce(final long contractNumber, final long nonce) {
                     context.payer(),
                     HandleContext.ConsensusThrottling.ON);
             childRecordBuilder.memo(LAZY_CREATION_MEMO);
-            if (childRecordBuilder.status() == SUCCESS) {
```

Comment:
I see the recordBuilder's evm address will be set when dispatching the synthetic transactions above.

Reply:
True, this information is already available in the synthetic transaction. 💯